### PR TITLE
ookla-speedtest: support darwin

### PR DESCRIPTION
**Pull request by cherryblossom000**
###### Description of changes

Support x86_64-darwin and aarch64-darwin.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/ookla-speedtest/default.nix [hunk 1]
```diff
@@ -4,7 +4,7 @@ let
   pname = "ookla-speedtest";
   version = "1.1.1";
```
Comment (cherryblossom000) [inline pkgs/tools/networking/ookla-speedtest/default.nix:5]: I’m not sure whether to change this to 1.1.1.84. macOS is the only platform where the latest version is 1.1.1.84; all the others is 1.1.1.

File: pkgs/tools/networking/ookla-speedtest/default.nix [hunk 2]
```diff
@@ -13,6 +13,11 @@ let
       url = "https://install.speedtest.net/app/cli/${pname}-${version}-linux-aarch64.tgz";
       sha256 = "sha256-J2pAhz/hw8okohWAwvxkqpLtNY/8bbYHGhPQOo1DH9k=";
     };
+    x86_64-darwin = fetchurl {
+      url = "https://install.speedtest.net/app/cli/${pname}-${version}.84-macosx-x86_64.tgz";
+      sha256 = "sha256-FT925OUCortHDH98O0uK+qUOuYuxHoYhb8sai5JnbpQ=";
+    };
+    aarch64-darwin = x86_64-darwin;
```
Comment (cherryblossom000) [inline pkgs/tools/networking/ookla-speedtest/default.nix:20]: Ookla provides a single universal macOS binary.

Comment (Kranzes) [reply pkgs/tools/networking/ookla-speedtest/default.nix:20]: That doesn't seem right. How is a single universal binary even possible?

Comment (shyim) [reply pkgs/tools/networking/ookla-speedtest/default.nix:20]: @Kranzes that normals on macOS. Both binaries can be bundled into one.

image

Comment (Kranzes) [reply pkgs/tools/networking/ookla-speedtest/default.nix:20]: Interesting...

**Review by Kranzes — APPROVED** [review]

File: pkgs/tools/networking/ookla-speedtest/default.nix (hunk 2 above)
Comment (cherryblossom000) [reply pkgs/tools/networking/ookla-speedtest/default.nix:20]:
```sh
$ file ./result/bin/speedtest
./result/bin/speedtest: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64Mach-O 64-bit executable x86_64] [arm64]
./result/bin/speedtest (for architecture x86_64):	Mach-O 64-bit executable x86_64
./result/bin/speedtest (for architecture arm64):	Mach-O 64-bit executable arm64
```

**Review by bjornfor — APPROVED** [review]

**Pull request merged**